Components/Model/Product.php

```diff
@@ -392,6 +393,55 @@ protected function buildCategoryPaths(Article $article, Shop $shop)
                 $paths[] = $helper->buildCategoryPath($category);
             }
         }
+        $isProductStreamsAllowed = Shopware()
+            ->Plugins()
+            ->Frontend()
+            ->NostoTagging()
+            ->Config()
+            ->get(NostoBootstrap::CONFIG_PRODUCT_STREAMS);
+        if ($isProductStreamsAllowed) {
+            $paths = $this->amendRelatedProductStreams($article, $paths);
+            $paths = $this->amendRelatedProductSelectionStreams($article, $paths);
+        }
+        return $paths;
+    }
+
+    /**
+     * Add Product Streams to a given array of category paths
+     *
+     * @param Article $article
+     * @param array $paths
+     * @return array
+     */
+    public function amendRelatedProductStreams(Article $article, array $paths)
```

Comment:
Rename to `generateRelatedProductStreams` or make the method to directly push the product streams into categories